nulib/meadow
# Invalidate poster cache after poster generation

# Summary 

Fixes a bug where resetting a poster image does not update in the UI because the image has been cached. 

# Specific Changes in this PR

- Invalidate the Cloudfront cache for the poster key after poster image is generated
- Should be skipped in the dev/test environment
- Adds variables for `iiif_cloudfront_distribution_id` 
- Gives Meadow task permissions to create cloudfront invalidation

**Terraform changes applied on staging

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

- Pull changes from `nulib/tfvars`

Dev: 
- We don't cache in the dev environment so hard to test
- But you should see in the logs that it's skipping invalidation if you generate a new poster image

Staging/Prod:
- set a new poster image on a file set that already has a poster image
- wait a minute or so and refresh the page
- expect to see the new poster image reflected in the UI

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes
- [x] Terraform changes **Terraform changes applied on staging
  - [x] Adds/requires new or changed Terraform variables (tfvars)


# Tested/Verified
